app/service/sorting.go

```diff
@@ -0,0 +1,191 @@
+package service
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/France-ioi/AlgoreaBackend/app/database"
+)
+
+// FieldSortingParams represents sorting parameters for one field
+type FieldSortingParams struct {
+	// ColumnName is a DB column name (may contain a table name as a prefix, e.g. "groups.ID")
+	ColumnName string
+	// FieldType is one of "int64", "bool", "string"
+	FieldType string
+}
+
+// ApplySorting applies ordering and paging according to given accepted fields and sorting rules
+// taking into the account the URL parameters 'from.*'
+func ApplySorting(r *http.Request, query *database.DB, acceptedFields map[string]*FieldSortingParams,
```

Comment:
Shouldn't the name reflect that there is paging done as well?

Reply:
fixed